packages/core/src/App/Components/Layout/Header/toggle-menu-drawer.jsx

```diff
@@ -170,7 +171,14 @@ const ToggleMenuDrawer = React.forwardRef(
         }, [is_appstore, is_pre_appstore, account_status, should_allow_authentication]);
 
         const toggleDrawer = React.useCallback(() => {
-            setIsOpen(!is_open);
+            if (!is_open) setIsOpen(!is_open);
+            else {
+                setTransitionExit(true);
+                setTimeout(() => {
+                    setIsOpen(false);
+                    setTransitionExit(false);
+                }, 400);
+            }
```

Comment:
@iman-fs I think you can move this logic into the component itself to isolate it and also make it reusable, Have a look at how @mitra-deriv has done it [here](https://github.com/binary-com/deriv-com/pull/3871/files#diff-27e90ba3c5be58c9805bc09efd53f7e959b2bd1eaee812489e4b073939fc1660R106).

Reply:
sure man. I will check it